$NetBSD$

Avoid installing sq and unsq man pages since we do not install those
programs, and terminate the remaining recipe cleanly.

Honor and propagate the configured strip command and staging DESTDIR.  In
particular, pkgsrc sets stripping to a no-op on OpenVMS because the native
image format must not be processed by the GNV strip utility, and GNV make
does not propagate command-line variables through this recursive make.

Name compiler outputs explicitly because VSI C otherwise derives uppercase
object names from source files replaced by the native patch utility.

--- Makefile.orig
+++ Makefile
@@ -301,7 +301,7 @@
 .c.o:
 	@. ./config.sh; \
 	  set -x; \
-	  $$CC $$CFLAGS -c $<
+	  $$CC $$CFLAGS -c -o $@ $<
 
 #
 # The funny business with y_tab.c is necessary for MS-DOS systems,
@@ -312,8 +312,7 @@
 	  set -x; \
 	  $$YACC $<; \
 	  [ -f y_tab.c ]  ||  mv y.tab.c y_tab.c; \
-	  $$CC $$CFLAGS -c y_tab.c; \
-	  mv y_tab.o $@; \
+	  $$CC $$CFLAGS -c -o $@ y_tab.c; \
 	  rm -f y_tab.c
 
 all-languages:	munchable
@@ -336,7 +335,7 @@
 	@. ./config.sh; \
 	  set -x; \
 	  cd $(DESTDIR)$$BINDIR; \
-	  strip ispell$$EXEEXT; \
+	  $(STRIP) ispell$$EXEEXT; \
 	  chmod 755 ispell$$EXEEXT
 	@. ./config.sh; \
 	  set -x; \
@@ -373,7 +372,8 @@
 	@. ./config.sh; \
 	  set -x; \
 	  cd deformatters; \
-	  $(MAKE) EXEEXT=$$EXEEXT install
+	  $(MAKE) EXEEXT=$$EXEEXT STRIP="$(STRIP)" \
+	    DESTDIR="$(DESTDIR)" install
 
 install-dictbuild:
 	@. ./config.sh; \
@@ -392,7 +392,7 @@
 	@. ./config.sh; \
 	  set -x; \
 	  cd $(DESTDIR)$$BINDIR; \
-	  strip buildhash$$EXEEXT icombine$$EXEEXT ijoin$$EXEEXT; \
+	  $(STRIP) buildhash$$EXEEXT icombine$$EXEEXT ijoin$$EXEEXT; \
 	  chmod 755 buildhash$$EXEEXT icombine$$EXEEXT ijoin$$EXEEXT \
 	    munchlist findaffix tryaffix
 	@. ./config.sh; \
@@ -405,13 +405,10 @@
 	    rm -f fixispell-a$$MAN1EXT
 	@. ./config.sh; \
 	  set -x; \
-	  $$INSTALL sq.1 $(DESTDIR)$$MAN1DIR/sq$$MAN1EXT; \
 	  for m in buildhash munchlist findaffix tryaffix; do \
 	    echo ".so `basename $$MAN1DIR`/ispell$$MAN1EXT" \
 	      > $(DESTDIR)$$MAN1DIR/$$m$$MAN1EXT; \
-	  done; \
-	  echo ".so `basename $$MAN1DIR`/sq$$MAN1EXT" \
-	    > $(DESTDIR)$$MAN1DIR/unsq$$MAN1EXT 
+	  done
 	@. ./config.sh; \
 	  set -x; \
 	  cd $(DESTDIR)$$MAN1DIR; \
@@ -419,7 +416,7 @@
 	    munchlist$$MAN1EXT findaffix$$MAN1EXT tryaffix$$MAN1EXT
 
 install-languages:
-	$(MAKE) DESTDIR=$(DESTDIR) LANGUAGE_TARGET=install \
+	$(MAKE) DESTDIR="$(DESTDIR)" LANGUAGE_TARGET=install \
             SHELLDEBUG=$(SHELLDEBUG) \
 	  language-subdirs
 	. ./config.sh; \
@@ -468,7 +465,8 @@
 		set -x; \
 		cd languages/$$dir; \
 		eval $(MAKE) BUILD=$(BUILD) DBUILD=$(DBUILD) CBUILD=$(CBUILD) \
-		  SHELLDEBUG=$(SHELLDEBUG) "$$makeargs" $(LANGUAGE_TARGET) \
+		  DESTDIR="$(DESTDIR)" SHELLDEBUG=$(SHELLDEBUG) \
+		  "$$makeargs" $(LANGUAGE_TARGET) \
 		    ||  exit 1; \
 		)  ||  exit 1; \
 		LANGUAGES=`echo "$$LANGUAGES" \
@@ -625,7 +623,7 @@
 term.o:	term.c
 	@. ./config.sh; \
 	  echo $$CC $$CFLAGS -c term.c 1>&2; \
-	  $$CC $$CFLAGS -c term.c \
+	  $$CC $$CFLAGS -c -o $@ term.c \
 	    ||  (echo \
 		   "term.c wouldn't compile.  Try inverting USG in local.h" \
 		    1>&2; \
